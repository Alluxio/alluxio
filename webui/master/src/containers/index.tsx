/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

export {default as App} from './App/App';

export {default as Browse} from './pages/Browse/Browse';
export {default as Configuration} from './pages/Configuration/Configuration';
export {default as Data} from './pages/Data/Data';
export {default as Overview} from './pages/Overview/Overview';
export {default as MasterLogs} from './pages/MasterLogs/MasterLogs';
export {default as Workers} from './pages/Workers/Workers';
export {default as Metrics} from './pages/Metrics/Metrics';
