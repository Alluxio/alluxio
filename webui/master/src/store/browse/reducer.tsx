/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {Reducer} from 'redux';

import {BrowseActionTypes, IBrowseState} from './types';

export const initialBrowseState: IBrowseState = {
  data: {
    'accessControlException': '',
    'blockSizeBytes': '',
    'currentDirectory': {
      'absolutePath': '',
      'blockSizeBytes': '',
      'blocksOnTier': {},
      'creationTime': '',
      'fileLocations': [],
      'group': '',
      'id': 0,
      'inAlluxio': false,
      'inAlluxioPercentage': 0,
      'isDirectory': true,
      'mode': '',
      'modificationTime': '',
      'name': '',
      'owner': '',
      'persistenceState': '',
      'pinned': false,
      'size': ''
    },
    'currentPath': '',
    'debug': false,
    'fatalError': '',
    'fileBlocks': [],
    'fileData': '',
    'fileDoesNotExistException': '',
    'fileInfos': [],
    'highestTierAlias': '',
    'invalidPathError': '',
    'invalidPathException': '',
    'masterNodeAddress': '',
    'ntotalFile': 0,
    'pathInfos': [],
    'showPermissions': true,
    'viewingOffset': 0
  },
  errors: undefined,
  loading: false,
};

export const browseReducer: Reducer<IBrowseState> = (state = initialBrowseState, action) => {
  switch (action.type) {
    case BrowseActionTypes.FETCH_REQUEST:
      return {...state, loading: true};
    case BrowseActionTypes.FETCH_SUCCESS:
      return {...state, loading: false, data: action.payload.data, response: action.payload, errors: undefined};
    case BrowseActionTypes.FETCH_ERROR:
      return {...state, loading: false, errors: action.payload};
    default:
      return state;
  }
};
