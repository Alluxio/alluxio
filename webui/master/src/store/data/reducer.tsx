/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {Reducer} from 'redux';

import {DataActionTypes, IDataState} from './types';

export const initialDataState: IDataState = {
  data: {
    'fatalError': '',
    'fileInfos': [],
    'inAlluxioFileNum': 0,
    'masterNodeAddress': '',
    'permissionError': '',
    'showPermissions': true
  },
  errors: undefined,
  loading: false
};

export const dataReducer: Reducer<IDataState> = (state = initialDataState, action) => {
  switch (action.type) {
    case DataActionTypes.FETCH_REQUEST:
      return {...state, loading: true};
    case DataActionTypes.FETCH_SUCCESS:
      return {...state, loading: false, data: action.payload.data, response: action.payload, errors: undefined};
    case DataActionTypes.FETCH_ERROR:
      return {...state, loading: false, errors: action.payload};
    default:
      return state;
  }
};
