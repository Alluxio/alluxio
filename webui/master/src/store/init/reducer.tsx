/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an 'AS IS' basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {Reducer} from 'redux';

import {IInitState, InitActionTypes} from './types';

export const initialInitState: IInitState = {
  data: {
    'debug': false,
    'newerVersionAvailable': false,
    'proxyDownloadFileApiUrl': {
      'prefix': 'http://192.168.1.6:39999/api/v1/paths/',
      'suffix': '/download-file/'
    },
    'refreshInterval': 15000,
    'securityAuthorizationPermissionEnabled': false,
    'webFileInfoEnabled': false,
    'workerPort': 30000
  },
  errors: undefined,
  loading: false
};

export const initReducer: Reducer<IInitState> = (state = initialInitState, action) => {
  switch (action.type) {
    case InitActionTypes.FETCH_REQUEST:
      return {...state, loading: true};
    case InitActionTypes.FETCH_SUCCESS:
      return {...state, loading: false, data: action.payload.data, response: action.payload, errors: undefined};
    case InitActionTypes.FETCH_ERROR:
      return {...state, loading: false, errors: action.payload};
    default:
      return state;
  }
};
