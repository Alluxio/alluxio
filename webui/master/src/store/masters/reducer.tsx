/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import { Reducer } from 'redux';

import { IMastersState, MastersActionTypes } from './types';

export const initialMastersState: IMastersState = {
  data: {
    debug: false,
    lostMasterInfos: [],
    standbyMasterInfos: [],
    primaryMasterInfo: {
      address: {
        rpcPort: 0,
        host: '',
      },
      id: '',
      lastUpdatedTime: '',
      startTime: '',
      gainPrimacyTime: '',
      losePrimacyTime: '',
      lastCheckpointTime: '',
      journalEntriesSinceCheckpoint: 0,
      version: '',
      revision: '',
    },
  },
  errors: undefined,
  loading: false,
};

export const mastersReducer: Reducer<IMastersState> = (state = initialMastersState, action) => {
  switch (action.type) {
    case MastersActionTypes.FETCH_REQUEST:
      return { ...state, loading: true };
    case MastersActionTypes.FETCH_SUCCESS:
      return { ...state, loading: false, data: action.payload.data, response: action.payload, errors: undefined };
    case MastersActionTypes.FETCH_ERROR:
      return { ...state, loading: false, errors: action.payload };
    default:
      return state;
  }
};
