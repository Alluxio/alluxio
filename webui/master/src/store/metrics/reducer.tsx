/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {Reducer} from 'redux';

import {transformToNivoFormat} from '@alluxio/common-ui/src/utilities';
import {IMetricsState, MetricsActionTypes} from './types';
import {LineSerieData} from '@nivo/line';

export const initialMetricsState: IMetricsState = {
  data: {
    'cacheHitLocal': '',
    'cacheHitRemote': '',
    'cacheMiss': '0.00',
    'masterCapacityFreePercentage': 0,
    'masterCapacityUsedPercentage': 0,
    'masterUnderfsCapacityFreePercentage': 0,
    'masterUnderfsCapacityUsedPercentage': 0,
    'operationMetrics': {},
    'rpcInvocationMetrics': {},
    'timeSeriesMetrics': [],
    'totalBytesReadLocal': '',
    'totalBytesReadLocalThroughput': '',
    'totalBytesReadDomainSocket': '',
    'totalBytesReadDomainSocketThroughput': '',
    'totalBytesReadRemote': '',
    'totalBytesReadRemoteThroughput': '',
    'totalBytesReadUfs': '',
    'totalBytesReadUfsThroughput': '',
    'totalBytesWrittenAlluxio': '',
    'totalBytesWrittenAlluxioThroughput': '',
    'totalBytesWrittenDomainSocket': '',
    'totalBytesWrittenDomainSocketThroughput': '',
    'totalBytesWrittenUfs': '',
    'totalBytesWrittenUfsThroughput': '',
    'ufsOps': {},
    'ufsReadSize': {},
    'ufsWriteSize': {}
  },
  errors: undefined,
  loading: false
};

export const metricsReducer: Reducer<IMetricsState> = (state = initialMetricsState, action) => {
  switch (action.type) {
    case MetricsActionTypes.FETCH_REQUEST:
      return {...state, loading: true};
    case MetricsActionTypes.FETCH_SUCCESS:
      const timeSeriesMetrics: LineSerieData[] = [];
      action.payload.data.timeSeriesMetrics.map((item: any) => {
        // only push the latest 20 points of data
        timeSeriesMetrics.push({
          id: item.name,
          xAxisLabel: 'Time Stamp',
          yAxisLabel: 'Percent (%)',
          data: transformToNivoFormat(item.dataPoints.splice(0,24), 'timeStamp', 'value')
        });
      });
      action.payload.data.timeSeriesMetrics = timeSeriesMetrics;
      return {...state, loading: false, data: action.payload.data, response: action.payload, errors: undefined};
    case MetricsActionTypes.FETCH_ERROR:
      return {...state, loading: false, errors: action.payload};
    default:
      return state;
  }
};
