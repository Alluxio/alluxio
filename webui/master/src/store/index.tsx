/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {connectRouter, RouterState} from 'connected-react-router';
import {History} from 'history';
import {combineReducers} from 'redux';
import {all, fork} from 'redux-saga/effects';

import {initialRefreshState, refreshReducer} from '@alluxio/common-ui/src/store/refresh/reducer';
import {IRefreshState} from '@alluxio/common-ui/src/store/refresh/types';
import {initialLogsState, logsReducer} from '@alluxio/common-ui/src/store/logs/reducer';
import {logsSaga} from '@alluxio/common-ui/src/store/logs/sagas';
import {ILogsState} from '@alluxio/common-ui/src/store/logs/types';
import {browseReducer, initialBrowseState} from './browse/reducer';
import {browseSaga} from './browse/sagas';
import {IBrowseState} from './browse/types';
import {configReducer, initialConfigState} from './config/reducer';
import {configSaga} from './config/sagas';
import {IConfigState} from './config/types';
import {dataReducer, initialDataState} from './data/reducer';
import {dataSaga} from './data/sagas';
import {IDataState} from './data/types';
import {initialInitState, initReducer} from './init/reducer';
import {initSaga} from './init/sagas';
import {IInitState} from './init/types';
import {initialMetricsState, metricsReducer} from './metrics/reducer';
import {metricsSaga} from './metrics/sagas';
import {IMetricsState} from './metrics/types';
import {initialOverviewState, overviewReducer} from './overview/reducer';
import {overviewSaga} from './overview/sagas';
import {IOverviewState} from './overview/types';
import {initialWorkersState, workersReducer} from './workers/reducer';
import {workersSaga} from './workers/sagas';
import {IWorkersState} from './workers/types';

export interface IApplicationState {
  browse: IBrowseState;
  config: IConfigState;
  data: IDataState;
  init: IInitState;
  logs: ILogsState;
  metrics: IMetricsState;
  overview: IOverviewState;
  refresh: IRefreshState;
  router?: RouterState;
  workers: IWorkersState;
}

export const rootReducer = (history: History) => combineReducers<IApplicationState>({
  browse: browseReducer,
  config: configReducer,
  data: dataReducer,
  init: initReducer,
  logs: logsReducer,
  metrics: metricsReducer,
  overview: overviewReducer,
  refresh: refreshReducer,
  router: connectRouter(history),
  workers: workersReducer
});

export const rootSaga = function* () {
  yield all([
    fork(browseSaga),
    fork(configSaga),
    fork(dataSaga),
    fork(initSaga),
    fork(logsSaga),
    fork(metricsSaga),
    fork(overviewSaga),
    fork(workersSaga)
  ]);
};

export const initialState: IApplicationState = {
  browse: initialBrowseState,
  config: initialConfigState,
  data: initialDataState,
  init: initialInitState,
  logs: initialLogsState,
  metrics: initialMetricsState,
  overview: initialOverviewState,
  refresh: initialRefreshState,
  workers: initialWorkersState
};
