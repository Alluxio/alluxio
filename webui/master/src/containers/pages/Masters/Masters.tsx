/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import React from 'react';
import { connect } from 'react-redux';
import { Table } from 'reactstrap';
import { AnyAction, compose, Dispatch } from 'redux';

import { withErrors, withLoadingMessage, withFetchData } from '@alluxio/common-ui/src/components';
import { IMasterInfo } from '../../../constants';
import { IApplicationState } from '../../../store';
import { fetchRequest } from '../../../store/masters/actions';
import { IMasters } from '../../../store/masters/types';
import { IInit } from '../../../store/init/types';
import { createAlertErrors } from '@alluxio/common-ui/src/utilities';
import { ICommonState } from '@alluxio/common-ui/src/constants';

interface IPropsFromState extends ICommonState {
  initData: IInit;
  mastersData: IMasters;
}

interface IPropsFromDispatch {
  fetchRequest: typeof fetchRequest;
}

export type AllProps = IPropsFromState & IPropsFromDispatch;

export class MastersPresenter extends React.Component<AllProps> {
  public render(): JSX.Element {
    const { initData, mastersData } = this.props;

    return (
      <div className="masters-page">
        <div className="container-fluid">
          <div className="row">
            <div className="col-12">
              <h5>Leader Master</h5>
              <Table hover={true}>
                <thead>
                  <tr>
                    <th>Master Host</th>
                    <th>Master Port</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>{mastersData.leaderMasterInfo.address.host}</td>
                    <td>{mastersData.leaderMasterInfo.address.rpcPort}</td>
                  </tr>
                </tbody>
              </Table>
            </div>
          </div>
          <div className="row">
            <div className="col-12">
              <h5>Standby Masters</h5>
              <Table hover={true}>
                <thead>
                  <tr>
                    {initData.debug ? <th>[D]Master Id</th> : null}
                    <th>Master Host</th>
                    <th>Master Port</th>
                    <th>Last Heartbeat</th>
                  </tr>
                </thead>
                <tbody>
                  {mastersData.normalMasterInfos.map((masterInfo: IMasterInfo) => (
                    <tr key={masterInfo.id}>
                      {initData.debug ? <td>{masterInfo.id}</td> : null}
                      <td>{masterInfo.address.host}</td>
                      <td>{masterInfo.address.rpcPort}</td>
                      <td>{new Date(masterInfo.lastUpdatedTimeMs).toTimeString()}</td>
                    </tr>
                  ))}
                </tbody>
              </Table>
            </div>
          </div>
          <div className="row">
            <div className="col-12">
              <h5>Lost Masters</h5>
              <Table hover={true}>
                <thead>
                  <tr>
                    {initData.debug ? <th>[D]Master Id</th> : null}
                    <th>Master Host</th>
                    <th>Master Port</th>
                    <th>Last Heartbeat</th>
                  </tr>
                </thead>
                <tbody>
                  {mastersData.failedMasterInfos.map((masterInfo: IMasterInfo) => (
                    <tr key={masterInfo.id}>
                      {initData.debug ? <td>{masterInfo.id}</td> : null}
                      <td>{masterInfo.address.host}</td>
                      <td>{masterInfo.address.rpcPort}</td>
                      <td>{new Date(masterInfo.lastUpdatedTimeMs).toTimeString()}</td>
                    </tr>
                  ))}
                </tbody>
              </Table>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = ({ init, refresh, masters }: IApplicationState): IPropsFromState => ({
  initData: init.data,
  errors: createAlertErrors(init.errors !== undefined || masters.errors !== undefined),
  loading: init.loading || masters.loading,
  refresh: refresh.data,
  mastersData: masters.data,
  class: 'masters-page',
});

const mapDispatchToProps = (dispatch: Dispatch): { fetchRequest: () => AnyAction } => ({
  fetchRequest: (): AnyAction => dispatch(fetchRequest()),
});

export default compose(
  connect(
    mapStateToProps,
    mapDispatchToProps,
  ),
  withFetchData,
  withErrors,
  withLoadingMessage,
)(MastersPresenter) as typeof React.Component;
