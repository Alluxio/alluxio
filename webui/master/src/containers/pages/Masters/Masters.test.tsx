/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import { configure, shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createBrowserHistory, History, LocationState } from 'history';
import React from 'react';
import sinon from 'sinon';

import { initialState } from '../../../store';
import { initialInitState } from '../../../store/init/reducer';
import { AllProps, MastersPresenter } from './Masters';
import { routePaths } from '../../../constants';
import { createAlertErrors } from '@alluxio/common-ui/src/utilities';

configure({ adapter: new Adapter() });

describe('Masters', () => {
  let history: History<LocationState>;
  let props: AllProps;

  beforeAll(() => {
    history = createBrowserHistory({ keyLength: 0 });
    history.push(routePaths.masters);
    props = {
      initData: initialInitState.data,
      mastersData: initialState.masters.data,
      errors: createAlertErrors(false),
      loading: false,
      refresh: initialState.refresh.data,
      class: '',
      fetchRequest: sinon.spy(() => {}),
    };
  });

  afterEach(() => {
    sinon.restore();
  });

  describe('Shallow component', () => {
    let shallowWrapper: ShallowWrapper;

    beforeAll(() => {
      shallowWrapper = shallow(<MastersPresenter {...props} />);
    });

    it('Matches snapshot', () => {
      expect(shallowWrapper).toMatchSnapshot();
    });
  });
});
