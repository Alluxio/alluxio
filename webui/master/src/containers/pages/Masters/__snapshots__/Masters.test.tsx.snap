// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Masters Shallow component Matches snapshot 1`] = `
<div
  className="masters-page"
>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <h5>
          Leader Master
        </h5>
        <Table
          hover={true}
          responsiveTag="div"
          tag="table"
        >
          <thead>
            <tr>
              <th>
                Master Host
              </th>
              <th>
                Master Port
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td />
              <td>
                0
              </td>
            </tr>
          </tbody>
        </Table>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <h5>
          Standby Masters
        </h5>
        <Table
          hover={true}
          responsiveTag="div"
          tag="table"
        >
          <thead>
            <tr>
              <th>
                Master Host
              </th>
              <th>
                Master Port
              </th>
              <th>
                Last Heartbeat
              </th>
            </tr>
          </thead>
          <tbody />
        </Table>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <h5>
          Lost Masters
        </h5>
        <Table
          hover={true}
          responsiveTag="div"
          tag="table"
        >
          <thead>
            <tr>
              <th>
                Master Host
              </th>
              <th>
                Master Port
              </th>
              <th>
                Last Heartbeat
              </th>
            </tr>
          </thead>
          <tbody />
        </Table>
      </div>
    </div>
  </div>
</div>
`;
